src/facade/dragonfly_connection.cc

```diff
@@ -557,13 +561,17 @@ void Connection::HandleRequests() {
 
   FiberSocketBase* peer = socket_.get();
 #ifdef DFLY_USE_SSL
-  if (ctx_) {
+  if (ssl_ctx_) {
     const bool no_tls_on_admin_port = absl::GetFlag(FLAGS_no_tls_on_admin_port);
     if (!(IsPrivileged() && no_tls_on_admin_port)) {
-      unique_ptr<tls::TlsSocket> tls_sock = make_unique<tls::TlsSocket>(std::move(socket_));
-      tls_sock->InitSSL(ctx_);
-      FiberSocketBase::AcceptResult aresult = tls_sock->Accept();
-      SetSocket(tls_sock.release());
+      // Must be done atomically before the premption point in Accept so that at any
+      // point in time, the socket_ is defined.
+      {
+        unique_ptr<tls::TlsSocket> tls_sock = make_unique<tls::TlsSocket>(std::move(socket_));
+        tls_sock->InitSSL(ssl_ctx_);
+        SetSocket(tls_sock.release());
+      }
```

Comment:
Maybe a FiberAtomicGuard wouldn't hurt, so we verfiy that InitSSL never suspends. Actually, just exchaning the socket in two lines and calling Init() belog  might be simpler. Do we need the unique_ptr for unwind logic in case something throws?

Reply:
will add FiberAtomicGuard.  I prefer two statements, easier to read imho.